tests/test_apis.py

```diff
@@ -442,12 +434,6 @@ def tearDown(self):
         # Reset warning behavior back to default
         warnings.simplefilter('default')
 
-    def testNonUnicodeSource(self):
-        """ Test falure on non-unicode source text. """
-        if not PY3:
-            source = "foo".encode('utf-16')
-            self.assertRaises(UnicodeDecodeError, markdown.markdown, source)
-
```

Comment:
Let's leave this test in. It is entirely reasonable to expect that a user could pass in something other than `utf-8` source text and we should continue to return an appropriate error. I haven't checked this, but if the test was removed because it was failing, then we need to make some changes to the code to make it pass again.